R/error.R

```diff
@@ -10,20 +10,36 @@
 ##'   \code{ERROR} is used.  Ignored if \code{errors} is given.
 ##'
 ##' @param errors A named list of errors - use this to signal multiple
-##'   error conditions as key/value pairs
+##'   error conditions as key/value pairs.
 ##'
 ##' @param status_code The HTTP status code to use.  The default (400)
 ##'   means "bad request" which should be a reasonable catch-all for
 ##'   bad user data.
 ##'
+##' @param ... Additional named args to be included as fields in the
+##' error response JSON. The values must be in format ready for
+##' serialization to JSON using [jsonlite::toJSON()] i.e. any unboxing
+##' using [jsonlite::unbox()] needs to already have been done.
+##'
 ##' @return Nothing, as this function throws an error
 ##' @export
 pkgapi_stop <- function(message, code = "ERROR", errors = NULL,
-                        status_code = 400L) {
-  if (is.null(errors)) {
+                        status_code = 400L, ...) {
+  if (!is.null(errors)) {
+    ## Convert from key - value pairs to key - list(detail = value)
+    errors <- lapply(errors, function(error) list(detail = error))
+  } else {
     assert_scalar_character(message)
     assert_scalar_character(code)
-    errors <- set_names(list(message), code)
+    extra <- list(...)
```

Comment:
I am not sure about the validation here? Do we not want to allow any arbitrary data here, so long as it's named? In which case

```
content <- list(detail = message)
extra <- list(...)
if (length(extra) > 0) {
  assert_named(extra)
  content <- c(content, extra)
}
errors <- list()
errors[[code]] <- content # or errors <- set_names(list(content), code)
```

Reply:
Yeah I think we can allow any arbitrary data - let me update that